url4192078a2f_unsegnor_hipotetizador_tgz-pJ8-hipotetizador_MedirTiemposJ8/src/hipotetizador/Hipo.java

```diff
@@ -0,0 +1,1790 @@
+
+        //Recorrer la lista
+
+        return reglas;
+    }
+
+    public ArrayList<Regla> deducir(boolean[][] v) {
+        ArrayList<Regla> reglas = new ArrayList<>();
+
+        //Comprobar todas las combinaciones posibles de elementos en función de 
+
+        return reglas;
+    }
+
+    private ArrayList<Regla> reducir_reglas(ArrayList<Regla> reglas) {
+        //Ordenar las reglas por tamaño de antecedente de menor a mayor y después por tamaño del consecuente de mayor a menor
+        //si A->B y A->C y A->BC la última sobra?
+        //si A->B para cualquier X tal que AX->B esta regla no nos dice nada nuevo salvo que tenga mayor confianza que la primera
+        //luego nos interesan reglas de antecedente pequeño y consecuente grande
+
+        //si dos reglas tienen el mismo antecedente
+
+        return reglas;
+    }
+
+    /**
+     * Indica las entradas que se ven afectadas por las reglas que se le pasan
+     *
+     * @param reglas Array de reglas
+     * @param nentradas número de entradas máximo
+     * @param contar_antecedente indica si se tienen en cuenta los antecedentes
+     * de las reglas
+     * @param contar_consecuente indica si se tienen en cuenta los consecuentes
+     * de las reglas
+     * @return devuelve un vector de booleanos indicando si la entrada de la
+     * posción i se ve afectada por alguna regla o no
+     */
+    private boolean[] calcular_entradas_afectadas(ArrayList<Regla> reglas, int nentradas, boolean contar_antecedente, boolean contar_consecuente) {
+        boolean[] respuesta = new boolean[nentradas];
+
+        //Llenar de falses
+        Arrays.fill(respuesta, false);
+
+        //Recorrer las reglas y anotar las entradas a las que afectan
+        for (Regla r : reglas) {
+            if (contar_antecedente) {
+                for (Elemento a : r.getAntecedente().getElementos()) {
+                    if (!respuesta[a.getEntrada()]) {
+                        respuesta[a.getEntrada()] = true;
+                    }
+                }
+            }
+            if (contar_consecuente) {
+                for (Elemento a : r.getConsecuente().getElementos()) {
+                    if (!respuesta[a.getEntrada()]) {
+                        respuesta[a.getEntrada()] = true;
+                    }
+                }
+            }
+        }
+
+        return respuesta;
+    }
+
+    private int[] evaluar(boolean[] estado, ArrayList<Regla> reglas) {
+        int[] transformado = new int[estado.length];
+        //Convertir el estado en vector de int
+        for (int i = 0; i < estado.length; i++) {
+            transformado[i] = estado[i] ? 1 : 0;
+        }
+
+        return evaluar(transformado, reglas);
+    }
+
+    /**
+     * Evalúa un estado para devolve el estado siguiente según las reglas TODO
+     * En realidad debería recibir y devolver una ventana
+     *
+     * @param b
+     * @return
+     */
+    private int[] evaluar(int[] b, ArrayList<Regla> reglas) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 957-998, lines 1003-1027 into separate methods.